pkg/3scale/amp/operator/redis_options_provider.go

```diff
@@ -272,3 +271,37 @@ func (r *RedisOptionsProvider) setTopologySpreadConstraints() {
 		r.options.BackendRedisTopologySpreadConstraints = r.apimanager.Spec.Backend.RedisTopologySpreadConstraints
 	}
 }
+
+func (r *RedisOptionsProvider) systemRedisPodTemplateAnnotations() map[string]string {
+	annotations := make(map[string]string)
+	for k, v := range r.apimanager.Spec.System.RedisAnnotations {
+		annotations[k] = v
+	}
+	annotations["generationID"] = r.generateUniqueID()
+	return annotations
+}
+
+func (r *RedisOptionsProvider) backendRedisPodTemplateAnnotations() map[string]string {
+	annotations := make(map[string]string)
+	for k, v := range r.apimanager.Spec.Backend.RedisAnnotations {
+		annotations[k] = v
+	}
+	annotations["generationID"] = r.generateUniqueID()
+	return annotations
+}
+
+func (r *RedisOptionsProvider) generateUniqueID() string {
```

Comment:
We are retrieving resourceVersion here right? Not generating it

Reply:
yes, thank you, it was different implementation before. Function renamed . 
Done